app/src/common/shared/com/igalia/wolvic/VRBrowserActivity.java

```diff
@@ -728,17 +737,18 @@ public void onConfigurationChanged(Configuration newConfig) {
 
     @Override
     public void onSharedPreferenceChanged(SharedPreferences sharedPreferences, String key) {
-        try {
             if (key.equals(getString(R.string.settings_key_voice_search_service))) {
-                SpeechRecognizer speechRecognizer =
-                        SpeechServices.getInstance(this, SettingsStore.getInstance(this).getVoiceSearchService());
-                ((VRBrowserApplication) getApplication()).setSpeechRecognizer(speechRecognizer);
+                try {
+                    SpeechRecognizer speechRecognizer =
+                            SpeechServices.getInstance(this, SettingsStore.getInstance(this).getVoiceSearchService());
+                    ((VRBrowserApplication) getApplication()).setSpeechRecognizer(speechRecognizer);
+                } catch (Exception e) {
+                    ((VRBrowserApplication) getApplication()).setSpeechRecognizer(null);
+                    e.printStackTrace();
```

Comment:
Can we refactor this so it can be used also from the onCreate() method?

Reply:
Yes, good idea.

Reply:
Done.